### PR TITLE
feat: job filter option

### LINKED ISSUE #349: Add Filters in Job Search

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #349 

Changes: 
- Added Filter for Jobs

Video for the change:
https://drive.google.com/open?id=1-BObN3ZCCjmBYBthQcQxOBNZWJxk3K6O

### CODE REVIEW

**Comment by adityastic** [comment]
@aggarwalpulkit596 have a look

**Comment by aggarwalpulkit596** [comment]
LGTM

**Comment by adityastic** [comment]
> LGTM

Glad to hear it xD

**Comment by adityastic** [comment]
@aggarwalpulkit596 also can we merge this sir

**Comment by adityastic** [comment]
@aggarwalpulkit596 any updates on this ?

**Comment by aggarwalpulkit596** [comment]
Oh yes, you need to do this filtering on the server-side because there might be some jobs which are not available on local as we do not get all the jobs available on the web so you need to modify your logic accordingly.
Sorry Running Busy, couldn't reply earlier

**Comment by adityastic** [comment]
@aggarwalpulkit596 can you tell me what logic needs to be modified because this is directly applied on the jobs list which gets updated from the api, as soon as jobs are refreshing the filter too will. Can you tell me the server repo, I will let you know if I am able to make changes to it or not, Thanks

**Comment by aggarwalpulkit596** [comment]
I'm talking about that these result should be filtered from server-side rather than the client-side and currently, you're filtering these results on client-side with the pre-existing data

**Comment by adityastic** [comment]
@aggarwalpulkit596 yes true, so what should I change to fix this issue at the moment? Because that is the only possible thing from the client side right?

**Comment by aggarwalpulkit596** [comment]
Make an appropriate API call with the selected filters as query params

**Comment by adityastic** [comment]
@aggarwalpulkit596 Amazing, I've updated the PR. It now fetches the filtered jobs with 
``` @Query("filter[location][\$ilike][\$any][]") ```
please review

File: onlineapi/src/main/kotlin/com/codingblocks/onlineapi/api/OnlineJsonApi.kt [hunk 1]
```diff
@@ -187,6 +187,17 @@ interface OnlineJsonApi {
         @Query("sort") sort: String = "-postedOn"
     ): Call<ArrayList<Jobs>>
 
+    @GET("jobs")
```
Comment (aggarwalpulkit596) [inline onlineapi/src/main/kotlin/com/codingblocks/onlineapi/api/OnlineJsonApi.kt]: no need to make a separate method  for this update the existing method and use that only

**Comment by aggarwalpulkit596** [comment]
> @aggarwalpulkit596 Amazing, I've updated the PR. It now fetches the filtered jobs with
> `@Query("filter[location][\$ilike][\$any][]")`
> please review
> 
> Edit: Also the result is the same xD

It might be same for now as the portal is still in the development phase and there are not a lot of companies but the number will grow and the old method will not be an appropriate solution at that time

**Comment by adityastic** [comment]
> > @aggarwalpulkit596 Amazing, I've updated the PR. It now fetches the filtered jobs with
> > `@Query("filter[location][\$ilike][\$any][]")`
> > please review
> > Edit: Also the result is the same xD
> 
> It might be same for now as the portal is still in the development phase and there are not a lot of companies but the number will grow and the old method will not be an appropriate solution at that time

Yes Completely valid. Let me make the required changes

File: onlineapi/src/main/kotlin/com/codingblocks/onlineapi/api/OnlineJsonApi.kt (hunk 1 above)
Comment (adityastic) [reply onlineapi/src/main/kotlin/com/codingblocks/onlineapi/api/OnlineJsonApi.kt]: updated @aggarwalpulkit596

**Comment by adityastic** [comment]
Also @aggarwalpulkit596 , There are some bounties claimed by me. Please have a look at them too. The contest is about to end, please take a look

**Comment by adityastic** [comment]
@aggarwalpulkit596 i've claimed bounty for this, as soon as you come online let me know if you need any other changes, please also check my other claims

**Comment by adityastic** [comment]
@aggarwalpulkit596 plz check this pr, today is the last day of the competition. Plz also check the bounties left for me, I'd really appreciate it. Thanks

**Comment by aggarwalpulkit596** [comment]
Don't worry I'll do this and and any pending claims as well no new claims will be entertained but the existing can be done post 15th August as well.

**Comment by adityastic** [comment]
@aggarwalpulkit596 thank you so much

**Comment by adityastic** [comment]
@aggarwalpulkit596 please also give me feedback on this https://medium.com/p/b37c889414c6/edit

**Comment by adityastic** [comment]
@aggarwalpulkit596 please also give me feedback on this https://medium.com/@adityaofficialgupta/my-boss-2019-experience-b37c889414c6

**Comment by aggarwalpulkit596** [comment]
Sure will do that in few minutes

**Comment by adityastic** [comment]
Ok cool

**Pull request merged**